tests/functional/search-auth.js

```diff
@@ -198,6 +198,48 @@ registerSuite("Search (auth)", {
         .end();
     },
 
+    "Label have correct URL after search without results"() {
+      // load up a garbage search, so we can more easily detect when
+      // the search values we want are loaded.
+      return FunctionalHelpers.openPage(
+        this,
+        url(
+          "/issues",
+          "?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc"
+        ),
+        ".grid"
+      )
+        .findByCssSelector("#js-SearchForm-input")
+        .clearValue()
+        .click()
+        .type("noExpectedResult123")
+        .submit()
+        .getProperty("value")
+        .then(function(text) {
+          assert.equal(
+            text,
+            "noExpectedResult123",
+            "The q param populated the search input."
+          );
+        })
+        .end()
+        .findByCssSelector("#x-search-bar")
+        .submit()
+        .end()
+        .sleep(2000)
```

Comment:
Does this test work without the `sleep(2000)`? In theory, there's a default find timeout of 10 seconds.

Reply:
It does not as it looks like the test is loosing the scope through the redirect. The label element is not found.

Reply:
Weird, there isn't even a real redirect here, just some XHR requests. But if it works... seems good. :)